### PR TITLE
New Display Package Page: created flight and setup for implementation of new page 

### PR DESCRIPTION
**Pull request by sophiamfavro**
Created a flight for the new page design and created the necessary duplicate files for my changes. I also made all the necessary additions so those files would work if flight turned off. 

If you have any questions for why I am doing this look at the Display Package Page Redesign Engineering spec, comment, or reach out to me.

### CODE REVIEW

File: src/Bootstrap/less/theme/page-display-package-v2.less [hunk 1]
```diff
@@ -0,0 +1,327 @@
+.page-package-details-v2 {
```
Comment (loic-sharma) [inline src/Bootstrap/less/theme/page-display-package-v2.less:1]: This file is currently almost identical to the original [`page-display-package.less` file](https://raw.githubusercontent.com/NuGet/NuGetGallery/0df692cd08e75b884d1bd6b652e8cd6c02685bdc/src/Bootstrap/less/theme/page-display-package.less). This first line is the only difference:

```diff
- .page-package-details {
+ .page-package-details-v2 {
  ...
}
```

File: src/NuGetGallery.Services/Configuration/IFeatureFlagService.cs [hunk 2]
```diff
@@ -173,6 +173,11 @@ public interface IFeatureFlagService
         /// </summary>
         bool AreEmbeddedReadmesEnabled(User user);
 
+        /// <summary>
+        /// Weather the new design of the display package page is enabled.
```
Comment (loic-sharma) [inline src/NuGetGallery.Services/Configuration/IFeatureFlagService.cs:177]:
```suggestion
        /// Whether the new design of the display package page is enabled.
```

File: src/NuGetGallery/NuGetGallery.csproj [hunk 3]
```diff
@@ -1864,6 +1865,7 @@
     <Content Include="Views\Authentication\Register.cshtml" />
     <Content Include="Views\Packages\ContactOwners.cshtml" />
     <Content Include="Views\Packages\DisplayPackage.cshtml" />
+	<Content Include="Views\Packages\DisplayPackageV2.cshtml" />
```
Comment (loic-sharma) [inline src/NuGetGallery/NuGetGallery.csproj:1868]: It looks like the spacing is off here - there's a tab instead of 4 spaces:

```suggestion
    <Content Include="Views\Packages\DisplayPackageV2.cshtml" />
```

File: src/NuGetGallery/Scripts/gallery/page-display-package-v2.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
+﻿$(function () {
```
Comment (loic-sharma) [inline src/NuGetGallery/Scripts/gallery/page-display-package-v2.js:1]: This file is currently completely identical to [`page-display-package.js`](https://raw.githubusercontent.com/NuGet/NuGetGallery/0df692cd08e75b884d1bd6b652e8cd6c02685bdc/src/NuGetGallery/Scripts/gallery/page-display-package.js)

File: src/NuGetGallery/Web.config [hunk 5]
```diff
@@ -321,7 +321,7 @@
   -->
   <system.web>
     <httpCookies requireSSL="true" httpOnlyCookies="true"/>
-    <compilation debug="false" targetFramework="4.7.2">
+    <compilation debug="true" targetFramework="4.7.2">
```
Comment (loic-sharma) [inline src/NuGetGallery/Web.config:324]: Please undo the changes in this file. We have a security tool that complains if these values change. See [this conversation for more information](https://teams.microsoft.com/l/message/19:d10cd9192de2492986fba3baeea3093c@thread.skype/1621874948124?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=ad34ac55-6785-4e7c-82c3-7ba405d30c3c&parentMessageId=1621874649401&teamName=NuGet&channelName=Server&createdTime=1621874948124).

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 6]
```diff
@@ -0,0 +1,1234 @@
...
+        <text><b>Please note:</b> The last successfully published symbols package is still available for debugging and download.</text>
+    }
+}
+
+@* The following two helpers must be on a single line each so no extra whitespace is introduced in the expression when rendered. *@
+@* Helpers themselves are needed not to introduce that extra whitespce, which happens if they are inlined. *@
+@helper MakeLicenseLink(CompositeLicenseExpressionSegment segment) {<a href="@LicenseExpressionRedirectUrlHelper.GetLicenseExpressionRedirectUrl(segment.Value)" aria-label="License @segment.Value">@segment.Value</a>}
+@helper MakeLicenseSpan(CompositeLicenseExpressionSegment segment) {<span>@segment.Value</span>}
+
+<section role="main" class="container main-container page-package-details-v2">
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:244]: For folks reviewing, this file is intended to be almost identical to the original [`DisplayPackage.cshtml` file](https://github.com/NuGet/NuGetGallery/blob/0df692cd08e75b884d1bd6b652e8cd6c02685bdc/src/NuGetGallery/Views/Packages/DisplayPackage.cshtml). This line is the only difference:

```diff
...
-<section role="main" class="container main-container page-package-details">
+<section role="main" class="container main-container page-package-details-v2">
...
</section>
    ...
```

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 7]
```diff
@@ -0,0 +1,1234 @@
...
+            }
+
+            @if (Model.Deleted)
+            {
+                @ViewHelpers.AlertDanger(
+                        @<text>
+                            <strong>This package has been deleted from the gallery.</strong> It is no longer available
+                            for install/restore.
+                        </text>
+)
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:333]: Hey it looks like the spacing here got messed up by the code formatter. Please undo the changes here:

```suggestion
                if (Model.ValidatingTooLong)
                {
                    @ViewHelpers.AlertWarning(
                        @<text>
                            <strong>Package validation is taking longer than expected.</strong> This is not normal, but we are on it!
                        </text>
                    )
                }
            }

            @if (Model.FailedValidation)
            {
                @ViewHelpers.AlertDanger(
                    @<text>
                        <strong>Package publishing failed.</strong> This package could not be published due to the following reason(s):
                        <ul class="failed-validation-alert-list">
                            @foreach (var issue in Model.PackageValidationIssues)
                            {
                                <li>@Html.Partial("_ValidationIssue", issue)</li>
                            }
                        </ul>
                        @if (!Model.PackageValidationIssues.Any(i => i.IssueCode == ValidationIssueCode.Unknown))
                        {
                            var issuePluralString = Model.PackageValidationIssues.Count > 1 ? "all the issues" : "the issue";
                            <text>Once you've fixed @issuePluralString with your package, you can reupload it with the same ID and version.</text>
                        }
                        else
                        {
                            <text>Please contact <a href="mailto:support@nuget.org">support@nuget.org</a> to help fix your package.</text>
                        }
                    </text>)
            }

            @if (Model.Deleted)
            {
                @ViewHelpers.AlertDanger(
                    @<text>
                        <strong>This package has been deleted from the gallery.</strong> It is no longer available
                        for install/restore.
                    </text>
                )
            }
```

**Comment by loic-sharma** [comment]
⚠TODO: Let's add a test on which view was returned depending on the feature flag:

* If the feature flag is off, `DisplayPackage` view should be returned
* If the feature flag is on, `DisplayPackageV2` view should be return.

**Review by agr — APPROVED** [review]

File: src/NuGetGallery/NuGetGallery.csproj [hunk 8]
```diff
@@ -2317,6 +2318,9 @@
   <ItemGroup>
     <Content Include="Views\Packages\_DisplayPackageVulnerabilities.cshtml" />
   </ItemGroup>
+  <ItemGroup>
+    <Content Include="Views\Packages\DisplayPackageV2.cshtml" />
```
Comment (loic-sharma) [inline src/NuGetGallery/NuGetGallery.csproj:2322]: Don't forget to move this content include into the `ItemGroup` for the `.cshtml` files up here:

https://github.com/NuGet/NuGetGallery/blob/c4a3a4228b141c93f1194e899ed8f2a15af048c6/src/NuGetGallery/NuGetGallery.csproj#L1867

File: src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml [hunk 9]
```diff
@@ -0,0 +1,1242 @@
+﻿@using NuGet.Services.Validation
```
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:1]: It looks like this file still have a bunch of unintentional whitespace changes. See: https://www.diffchecker.com/DWpR7dum

Let's fix this file together during our sync this afternoon :)

Comment (loic-sharma) [reply src/NuGetGallery/Views/Packages/DisplayPackageV2.cshtml:1]: Fixed!
